docs/docs/tutorials/workflows/index.md

```diff
@@ -0,0 +1,1273 @@
+            return "Pass"
+        return "Fail"
+
+
+    def improve_joke(state: State):
+        """Second LLM call to improve the joke"""
+
+        msg = llm.invoke(f"Make this joke funnier by adding wordplay: {state['joke']}")
+        return {"improved_joke": msg.content}
+
+
+    def polish_joke(state: State):
+        """Third LLM call for final polish"""
+
+        msg = llm.invoke(f"Add a surprising twist to this joke: {state['improved_joke']}")
+        return {"final_joke": msg.content}
+
+
+    # Build workflow
+    workflow = StateGraph(State)
+
+    # Add nodes
+    workflow.add_node("generate_joke", generate_joke)
+    workflow.add_node("improve_joke", improve_joke)
+    workflow.add_node("polish_joke", polish_joke)
+
+    # Add edges to connect nodes
+    workflow.add_edge(START, "generate_joke")
+    workflow.add_conditional_edges(
+        "generate_joke", check_punchline, {"Pass": "improve_joke", "Fail": END}
+    )
+    workflow.add_edge("improve_joke", "polish_joke")
+    workflow.add_edge("polish_joke", END)
+
+    # Compile
+    chain = workflow.compile()
+
+    # Show workflow
+    display(Image(chain.get_graph().draw_mermaid_png()))
+
+    # Invoke
+    state = chain.invoke({"topic": "cats"})
+    print("Initial joke:")
+    print(state["joke"])
+    print("\n--- --- ---\n")
+    if "improved_joke" in state:
+        print("Improved joke:")
+        print(state["improved_joke"])
+        print("\n--- --- ---\n")
+
+        print("Final joke:")
+        print(state["final_joke"])
+    else:
+        print("Joke failed quality gate - no punchline detected!")
+    ```
+
+    **LangSmith Trace**
+
+    https://smith.langchain.com/public/a0281fca-3a71-46de-beee-791468607b75/r
+
+    **Resources:**
+
+    **LangChain Academy**
+
+    See our lesson on Prompt Chaining [here](https://github.com/langchain-ai/langchain-academy/blob/main/module-1/chain.ipynb).
+
+=== "Functional API (beta)"
+
+    ```python
+    from langgraph.func import entrypoint, task
+
+
+    # Tasks
+    @task
+    def generate_joke(topic: str):
+        """First LLM call to generate initial joke"""
+        msg = llm.invoke(f"Write a short joke about {topic}")
+        return msg.content
+
+
+    def check_punchline(joke: str):
+        """Gate function to check if the joke has a punchline"""
+        # Simple check - does the joke contain "?" or "!"
+        if "?" in joke or "!" in joke:
+            return "Fail"
+
+        return "Pass"
+
+
+    @task
+    def improve_joke(joke: str):
+        """Second LLM call to improve the joke"""
+        msg = llm.invoke(f"Make this joke funnier by adding wordplay: {joke}")
+        return msg.content
+
+
+    @task
+    def polish_joke(joke: str):
+        """Third LLM call for final polish"""
+        msg = llm.invoke(f"Add a surprising twist to this joke: {joke}")
+        return msg.content
+
+
+    @entrypoint()
+    def workflow(topic: str):
+        original_joke = generate_joke(topic).result()
```

Comment:
This is not equivalent to the graph above

Reply:
yup need to update condition in the original one, good catch